src/NUnitFramework/framework/Constraints/RangeConstraint.cs

```diff
@@ -72,7 +87,7 @@ public override ConstraintResult ApplyTo<TActual>(TActual actual)
         {
             if ( from == null || to == null || actual == null)
                 throw new ArgumentException( "Cannot compare using a null reference", "actual" );
-
+            CompareFromAndTo();
```

Comment:
I would expect exactly that, because the comparer is never set. IOW you can never put the constraint into an invalid state, consistent with the constructor check.
```c#
Assert.DoesNotThrow(()=> test.ApplyTo(7));
Assert.Throws<ArgumentException>(() => test.Using(rComparer));
Assert.DoesNotThrow(()=> test.ApplyTo(7));
```